flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -108,6 +111,11 @@
 
 	private final Options ALL_OPTIONS;
 
+	private static final String fileName = "yarn-app.ini";
+	private static final String cookieKey = "secureCookie";
```

Comment:
Good point. The yarn properties file should not be in the temp folder. Could you change that? We could do that in a separate commit.

If you want to store multiple application states, then let's change the layout of the yarn properties file. The goal should be to have only one configuration file and format. Could you do that in a separate commit? It shouldn't be too much work. Let me know if that would be possible.